MetaMask/metamask-extension
# fix: number formatting on swap + send tx detail

## **Description**

Fixes an issue when rendering details for a swap + send transaction.

Token amounts less than 1 with >= 6 leading zeroes were formatted in scientific notation.

`toString` and `toPrecision` both render scientific in that case, so this PR uses `toFixed` instead.

[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/26029?quickstart=1)

## **Related issues**

Fixes: https://github.com/MetaMask/metamask-extension/issues/25876

## **Manual testing steps**

1. Do a swap + send transaction with a source and/or destination amount < .000001
2. Click activity tab
3. Click on the transaction
4. Source and destination amounts should not be in scientific notation

## **Screenshots/Recordings**



### **Before**
<img width="340" alt="Screenshot 2024-07-22 at 4 09 49 PM" src="https://github.com/user-attachments/assets/64f2dd69-2ffe-4a15-8e59-63f519b416b8">


### **After**

<img width="344" alt="Screenshot 2024-07-22 at 4 14 20 PM" src="https://github.com/user-attachments/assets/517294b0-1ae5-49d1-82a5-e46a1345daa8">


## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
